framework/test-vectors/README.md

```diff
@@ -0,0 +1,113 @@
+[//]: # "Copyright Amazon.com Inc. or its affiliates. All Rights Reserved."
+[//]: # "SPDX-License-Identifier: CC-BY-SA-4.0"
+
+## Summary
+
+These test vectors are intended to be used to validate interoperability
+across implementations of clients in a runtime,
+for the AWS Crypto Tools AWS Cryptographic Material Providers Library (MPL).
+
+They are composed of JSON manifest files that define one or more test cases,
+including sufficient information to process each test case.
+These manifest files can also identify additional resources needed for a given test case.
+These resources will be identified with a URI.
+If identifying a local file, the URI will be a relative path from the manifest file's
+parent directory to the target file.
+
+Every manifest type definition must include a unique type name that will be used by
+test vector handlers to identify the manifest type.
+
+See https://github.com/awslabs/aws-crypto-tools-test-vector-framework for additional information.
+This framework is much of the motivation for this work.
+
+The manifest is meant to produce encryption and decryption materials that every
+runtime implementation can process and produce the desired output for that test vector.
```

Comment:
Is it worth further specifying "desired output" further? For example, this could be success or failure, if it's success, the byte values match, if it's failure, there's some kind of exception, etc. The motivation being to make it clear it's possible to write test vectors for failure cases.

Reply:
How would you suggest doing this? cc @texastony. Like concretely do you have a suggestion?

Reply:
Two Options:
1. An Expectation of the Class of Failure
2. An Expectation on the content of the Failure Message

Expectations on the content of the failure Message can be fragile.
This [GHI from ESDK-Python](https://github.com/aws/aws-encryption-sdk-python/issues/644) is caused by too strict of a regular expression on the content of error messages.

So this is a tricky line.
It would be easier to walk it if all of our Specifications had stronger definitions on Error Classes or their content.
But at least for now, we could basket the failures:
- Error from Primitives
- Error from KMS
- Error from DDB